### PR TITLE
MRL bugfixes

### PR DESCRIPTION
**Pull request by WernerVdM97**
Summary of this change:
- Fixed erroneous update of pre-harvest child result where `pre_harvest_id IS nil` (Applicable to clients with `capture_mrl_result_per_chemicial = false`.
- Rename date time (..._at) fields to date (..._date) fields:
  - fruit_received_at
  - result_received_at
  - sample_submitted_at
- Prevent the user from selecting a date later than the current one for the above-mentioned fields.

### CODE REVIEW

File: db/migrations/202302201000_rename_mrl_results_columns_from_at_to_date.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Sequel.migration do
+  up do
+    alter_table(:mrl_results) do
```
Comment (jcsjcs) [inline db/migrations/202302201000_rename_mrl_results_columns_from_at_to_date.rb:3]: These should also be changed to date types.
Try
```ruby
set_column_type :fruit_received_at, Date
```
Or
```sql
ALTER TABLE public.mrl_results ALTER COLUMN fruit_received_at TYPE date USING fruit_received_at::date;
```

File: lib/quality/interactors/mrl_result_interactor.rb [hunk 2]
```diff
@@ -73,9 +73,9 @@ def delivery_mrl_result_attrs(delivery_id)
       attrs = repo.mrl_result_attrs_for(delivery_id, arr)
 
       success_response('Ok', attrs.merge({ rmt_delivery_id: delivery_id,
-                                           fruit_received_at: attrs[:date_delivered],
-                                           sample_submitted_at: Time.now,
-                                           result_received_at: nil }))
+                                           fruit_received_date: attrs[:date_delivered],
+                                           sample_submitted_date: Time.now,
```
Comment (jcsjcs) [inline lib/quality/interactors/mrl_result_interactor.rb:77]: `Date.today`

File: lib/quality/test/factories/mrl_result_factory.rb [hunk 3]
```diff
@@ -28,9 +28,9 @@ def create_mrl_result(opts = {}) # rubocop:disable Metrics/CyclomaticComplexity,
         pre_harvest_result: true,
         post_harvest_result: false,
         active: true,
-        fruit_received_at: '2010-01-01 12:00',
-        sample_submitted_at: '2010-01-01 12:00',
-        result_received_at: '2010-01-01 12:00',
+        fruit_received_date: '2010-01-01 12:00',
+        sample_submitted_date: '2010-01-01 12:00',
+        result_received_date: '2010-01-01 12:00',
```
Comment (jcsjcs) [inline lib/quality/test/factories/mrl_result_factory.rb:33]: Remove the time portions from these strings.

File: lib/quality/test/interactors/test_mrl_result_interactor.rb [hunk 4]
```diff
@@ -108,9 +108,9 @@ def mrl_result_attrs
         mrl_sample_passed: false,
         pre_harvest_result: false,
         post_harvest_result: false,
-        fruit_received_at: '2010-01-01 12:00',
-        sample_submitted_at: '2010-01-01 12:00',
-        result_received_at: '2010-01-01 12:00',
+        fruit_received_date: '2010-01-01 12:00',
+        sample_submitted_date: '2010-01-01 12:00',
+        result_received_date: '2010-01-01 12:00',
```
Comment (jcsjcs) [inline lib/quality/test/interactors/test_mrl_result_interactor.rb:113]: Remove the time portions from these strings.

File: lib/quality/test/task_permission_checks/test_mrl_result.rb [hunk 5]
```diff
@@ -29,9 +29,9 @@ def entity(attrs = {})
         mrl_sample_passed: false,
         pre_harvest_result: false,
         post_harvest_result: false,
-        fruit_received_at: '2010-01-01 12:00',
-        sample_submitted_at: '2010-01-01 12:00',
-        result_received_at: '2010-01-01 12:00',
+        fruit_received_date: '2010-01-01 12:00',
+        sample_submitted_date: '2010-01-01 12:00',
+        result_received_date: '2010-01-01 12:00',
         active: true,
```
Comment (jcsjcs) [inline lib/quality/test/task_permission_checks/test_mrl_result.rb:35]: Remove the time portions from these strings.

File: lib/quality/ui_rules/mrl_result_rule.rb [hunk 6]
```diff
@@ -87,9 +87,9 @@ def set_show_fields # rubocop:disable Metrics/AbcSize
       fields[:pre_harvest_result] = { renderer: :label, as_boolean: true }
       fields[:post_harvest_result] = { renderer: :label, as_boolean: true }
       fields[:active] = { renderer: :label, as_boolean: true }
-      fields[:fruit_received_at] = { renderer: :label, format: :without_timezone_or_seconds }
-      fields[:sample_submitted_at] = { renderer: :label, format: :without_timezone_or_seconds }
-      fields[:result_received_at] = { renderer: :label, format: :without_timezone_or_seconds }
+      fields[:fruit_received_date] = { renderer: :label, with_value: @form_object.fruit_received_date.strftime('%Y-%m-%d') }
+      fields[:sample_submitted_date] = { renderer: :label, with_value: @form_object.sample_submitted_date.strftime('%Y-%m-%d') }
+      fields[:result_received_date] = { renderer: :label, with_value: @form_object.result_received_date.strftime('%Y-%m-%d') }
```
Comment (jcsjcs) [inline lib/quality/ui_rules/mrl_result_rule.rb:92]: The `with_value` should not be required when the columns are no longer `Time`s.

File: lib/quality/ui_rules/mrl_result_rule.rb [hunk 7]
```diff
@@ -196,15 +196,18 @@ def common_fields # rubocop:disable Metrics/AbcSize
         mrl_sample_passed: { renderer: :checkbox },
         pre_harvest_result: { renderer: :hidden },
         post_harvest_result: { renderer: :hidden },
-        fruit_received_at: { renderer: :input,
-                             subtype: :date,
-                             required: true },
-        sample_submitted_at: { renderer: :input,
+        fruit_received_date: { renderer: :input,
                                subtype: :date,
+                               maxvalue: Time.now.strftime('%Y-%m-%d'),
```
Comment (jcsjcs) [inline lib/quality/ui_rules/mrl_result_rule.rb:201]: `Date.today` for this and the others.

File: lib/quality/ui_rules/mrl_result_rule.rb [hunk 8]
```diff
@@ -255,9 +258,10 @@ def set_print_mrl_labels_fields
     end
 
     def set_mrl_result_details(columns = nil, display_columns = 3)
+      hash = @form_object.map.to_h { |k, v| %i[fruit_received_date sample_submitted_date result_received_date].include?(k) ? [k, v.strftime('%Y-%m-%d')] : [k, v] }
```
Comment (jcsjcs) [inline lib/quality/ui_rules/mrl_result_rule.rb:261]: Should not be required when you change the column type to `Date`.

File: lib/quality/ui_rules/mrl_result_rule.rb [hunk 9]
```diff
@@ -291,9 +296,9 @@ def make_new_form_object
       @form_object = new_form_object_from_struct(QualityApp::MrlResult,
                                                  merge_hash: { pre_harvest_result: @options[:attrs][:pre_harvest_result],
                                                                post_harvest_result: @options[:attrs][:post_harvest_result],
-                                                               fruit_received_at: Time.now,
-                                                               sample_submitted_at: Time.now,
-                                                               result_received_at: nil })
+                                                               fruit_received_date: Time.now,
```
Comment (jcsjcs) [inline lib/quality/ui_rules/mrl_result_rule.rb:299]: `Date.today`.

File: lib/quality/validations/mrl_result_schema.rb [hunk 10]
```diff
@@ -20,8 +20,8 @@ module QualityApp
     required(:num_active_ingredients).maybe(:integer)
     required(:pre_harvest_result).maybe(:bool)
     required(:post_harvest_result).maybe(:bool)
-    required(:fruit_received_at).filled(:time)
-    required(:sample_submitted_at).filled(:time)
+    required(:fruit_received_date).filled(:time)
```
Comment (jcsjcs) [inline lib/quality/validations/mrl_result_schema.rb:23]: `filled(:date)` for all these fields.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

File: lib/quality/ui_rules/mrl_result_rule.rb (hunk 9 above)
Comment (jcsjcs) [reply lib/quality/ui_rules/mrl_result_rule.rb:297]: Still Time...

**Review by jcsjcs — CHANGES_REQUESTED** [review]

File: lib/quality/ui_rules/mrl_result_rule.rb (hunk 9 above)
Comment (WernerVdM97) [reply lib/quality/ui_rules/mrl_result_rule.rb]: Apologies, I have been too rushed with this. I fixed it now and double-checked the interactor as well.

**Pull request merged**